connection.go

```diff
@@ -711,6 +725,15 @@ func (conn *Connection) reader(r *bufio.Reader, c net.Conn) {
 			}
 		} else {
 			if fut = conn.fetchFuture(resp.RequestId); fut != nil {
+				if fut.ctx != nil {
+					select {
+					case <-fut.ctx.Done():
+						fut.SetError(ContextDoneErr)
+						conn.markDone(fut)
+						return
+					default:
+					}
+				}
```

Comment:
We need the same checks in `Connection.writer()` too.

Reply:
Added.